docs/coding-conventions.md

```diff
@@ -0,0 +1,297 @@
+
+- Use constants if you would repeat hardcoded values otherwise.
+- Keep them private if they are used only in a single package.
+- Carefully consider if you should make a constant exported.
+
+### Bad example
+
+```go
+func getImportantAnnotation(obj metav1.ObjectMeta) string {
+    return obj.GetAnnotations()["kubevirt.io/my-annotation"]
+}
+
+func setImportantAnnotation(obj metav1.ObjectMeta, val string) {
+    obj.GetAnnotations()["kubevirt.io/my-annotation"] = val
+}
+```
+
+### Good example
+
+```go
+const annotationKey = "kubevirt.io/my-annotation"
+
+func getImportantAnnotation(obj metav1.ObjectMeta) string {
+    return obj.GetAnnotations()[annotationKey]
+}
+
+func setImportantAnnotation(obj metav1.ObjectMeta, val string) {
+    obj.GetAnnotations()[annotationKey] = val
+}
+```
+
+# Uniform import order and naming
+
+- Use the following import order with one block per item:
+    - Golang standard libraries
+    - Ginkgo / Gomega imports (only in test files)
+    - Third-party libraries
+    - Local packages
+- Use the following naming schemes for imports:
+    - `virtv1` for imports of `kubevirt.io/api`
+    - `metav1` for imports of `k8s.io/apimachinery/pkg/apis/meta/v1`
+    - `k8s` prefix for `k8s.io` imports on collision with other imports
+    - TODO: Add more
+- `make format` is able to help you with this.
+
+### Bad example
+
+```go
+import (
+    "context"
+    k8sv1 "k8s.io/api/core/v1"
+    "time"
+
+    ."github.com/onsi/ginkgo/v2"
+    ."github.com/onsi/gomega"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    v1 "kubevirt.io/api/core/v1"
+    "kubevirt.io/kubevirt/tests/clientcmd"
+
+    "kubevirt.io/kubevirt/tests/testsuite"
+)
+```
+
+### Good example
+
+```go
+import (
+    "context"
+    "time"
+
+    ."github.com/onsi/ginkgo/v2"
+    ."github.com/onsi/gomega"
+
+    k8sv1 "k8s.io/api/core/v1"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    virtv1 "kubevirt.io/api/core/v1"
+
+    "kubevirt.io/kubevirt/tests/clientcmd"
+    "kubevirt.io/kubevirt/tests/testsuite"
+)
+```
+
+## Short coding conventions (TLDR)
+
+Here are some important but short conventions. If you want to learn more
+about conventions in detail take a look at the [detailed
+coding conventions](#detailed-coding-conventions).
+
+- [Write elegant, cohesive and easily readable code](#write-elegant-cohesive-and-easily-readable-code)
+- [Avoid nesting and complexity by using early returns](#avoid-nesting-and-complexity-by-using-early-returns)
+- [Use the same coding style throughout a file](#use-the-same-coding-style-throughout-a-file)
+- [Avoid repetition of hardcoded values](#avoid-repetition-of-hardcoded-values)
+- [Uniform import order and naming](#uniform-import-order-and-naming)
+- Prefert to inline variable assignments.
+    - For example: Inline `err` checks in if-statements.
+- Use switch-cases to avoid long if/else if/else statements.
+- Isolate code.
+    - Use interfaces instead of bare structs.
```

Comment:
Can you help making it more specific? e.g. also giving examples?

Reply:
I was lazy to suggest something, sorry.
Maybe:
> Use objects and polymorphism (through interfaces)
>   - Make sure to use interfaces to express behavior and not clone all object methods.
>   - Define interfaces at the usage, not together with the concrete object definition.
>   - For more information, see [here](https://go.dev/wiki/CodeReviewComments#interfaces)

You could also ref https://ehaas.net/blog/the-go-interface/ , but it may be considered opinionated and less trustful than the wiki one.

Reply:
Thanks, updated it and added the link to the end of the doc.